### DISCUSSION TITLE
Read data

### DISCUSSION THREAD

**Discussion by 19807311** [teamalgoritma/community#399] [Q&A] [answered]
halo, mau bertanya:
1. saat memanggil data, pakai read.csv yg keluar hanya satu variabel (semua kolom gabung. saya pakai read.csv2 baru bisa terpisah2. Bedanya apa?

2. Boleh bantu cara baca data bank.csv untuk variabel campaign, pdays, previous, duration. Pdyas ada minusnya ?

![pdays campaign previous](https://user-images.githubusercontent.com/114885734/195965888-2ba73a91-0dab-46fa-8258-5c65c06c1f0b.png)
![Screenshot_read data](https://user-images.githubusercontent.com/114885734/195965902-07c4189d-e8f1-45c7-82c1-5b842b7cca20.png)

**Comment by onlyphantom** [comment, marked as answer]
From the documentation, `read.csv` and `read.csv2` are **almost the same function** (hampir sama). The key difference is the `sep` parameter and decimals (one uses`.` for decimals, but another uses `,`):

```
> read.csv
function (file, header = TRUE, sep = ",", quote = "\"", dec = ".", 
    fill = TRUE, comment.char = "", ...) 

> read.csv2
function (file, header = TRUE, sep = ";", quote = "\"", dec = ",", 
    fill = TRUE, comment.char = "", ...) 
```

Terlihat `read.csv` memakai `,` (comma) untuk memisahkan columns nya, tetapi `read.csv2` memakai `;` (semicolon). Kebetulan `bank.csv` yang Ibu pakai itu pakai `;` secara delimiter, oleh karena itu, kalau mau tetap pakai `read.csv` tinggal di ganti aja separatornya:

```
bank <- read.csv("bank.csv", sep=";")
```

Good luck!

**Reply by 19807311** [reply]
I see.. Thank you, Pak Sam
